service/src/main/java/greencity/service/EventServiceImpl.java

```diff
@@ -137,14 +127,105 @@ public PageableAdvancedDto<EventDto> getAll(Pageable page, Principal principal)
     }
 
     @Override
-    public PageableAdvancedDto<EventDto> getAllUserEvents(Pageable page, String email) {
+    public PageableAdvancedDto<EventDto> getAllUserEvents(
+        Pageable page, String email, String userLatitude, String userLongitude, String eventType) {
         User attender = modelMapper.map(restClient.findByEmail(email), User.class);
-        Page<Event> events = eventRepo.findAllByAttender(page, attender.getId());
-        PageableAdvancedDto<EventDto> eventDtos = buildPageableAdvancedDto(events);
-        setSubscribes(events, eventDtos, attender);
+        List<Event> events = sortUserEventsByEventType(eventType, attender, userLatitude, userLongitude);
+        Page<Event> eventPage = new PageImpl<>(events, page, events.size());
+        PageableAdvancedDto<EventDto> eventDtos = buildPageableAdvancedDto(eventPage);
+        setSubscribes(eventPage, eventDtos, attender);
         return eventDtos;
     }
 
+    private List<Event> sortUserEventsByEventType(
+        String eventType, User attender, String userLatitude, String userLongitude) {
+        if (eventType.equalsIgnoreCase("ONLINE")) {
+            return getOnlineUserEventsSortedByDate(attender);
+        }
+
+        if (eventType.equalsIgnoreCase("OFFLINE")) {
+            if (!userLatitude.isBlank() && !userLongitude.isBlank()) {
+                return getOfflineUserEventsSortedCloserToUserLocation(attender, userLatitude, userLongitude);
+            } else {
+                return getOfflineUserEventsSortedByDate(attender);
+            }
+        }
+        return eventRepo.findAllByAttender(attender.getId()).stream().sorted(getComparatorByDates())
+            .collect(Collectors.toList());
+    }
+
+    private List<Event> getOnlineUserEventsSortedByDate(User attender) {
+        return eventRepo.findAllByAttender(attender.getId()).stream()
+            .filter(event -> event.getDates().get(event.getDates().size() - 1).getOnlineLink() != null)
```

Comment:
What are you trying to do here?

Reply:
I tried to filter only online events, and they are online if event online link is not null. But I have changed this part of code into proposed by you method (getEventType()) in entity class Event. And filter works using this method now.